Mathlib/Algebra/Order/Field/Canonical/Defs.lean

```diff
@@ -18,10 +18,14 @@ variable {α : Type*}
 
 /-- A canonically linear ordered field is a linear ordered field in which `a ≤ b` iff there exists
 `c` with `b = a + c`. -/
-class CanonicallyLinearOrderedSemifield (α : Type*) extends CanonicallyOrderedCommSemiring α,
-  LinearOrderedSemifield α
+class CanonicallyLinearOrderedSemifield (α : Type*) extends LinearOrderedSemifield α,
+    CanonicallyOrderedCommSemiring α
 #align canonically_linear_ordered_semifield CanonicallyLinearOrderedSemifield
 
+attribute [instance 150] CanonicallyLinearOrderedSemifield.toLinearOrderedSemifield
+attribute [instance 100] CanonicallyLinearOrderedSemifield.toCanonicallyOrderedCommSemiring
+attribute [instance 0] CanonicallyLinearOrderedSemifield.toOrderBot
```

Comment:
Huh? Where did this instance even come from? This looks like a lean bug.

Reply:
I wonder if it would be better to write:
```
@[class]
structure CanonicallyLinearOrderedSemifield (α : Type*) extends LinearOrderedSemifield α,
    CanonicallyOrderedCommSemiring α

attribute [instance 100] CanonicallyLinearOrderedSemifield.toLinearOrderedSemifield
attribute [instance 100] CanonicallyLinearOrderedSemifield.toCanonicallyOrderedCommSemiring
```
that is, turn off automatic instance generation entirely and explicitly list the instances we actually want.

Reply:
Note also that if `Foo.toBar` wasn't already an instance, `attribute [instance 0]` will actually turn it into one!

Reply:
Using `@[class] structure` only here won't work. Because `CanonicallyOrderedCommSemiring` or its ancestors are still classes, there will still be instance parameters in `CanonicallyLinearOrderedSemifield.mk`, and lean will still generate extra instances. (I don't know what's going on here, but they seem relevant)

Reply:
Is there a reason you used `attribute [instance 0]` and not `attribute [-instance]`? Is the latter only local?

Reply:
Also, I opened https://github.com/leanprover/lean4/issues/2905

Reply:
yes, `attribute [-instance]` is only local. IIUC there is no way to globally remove an instance.